src/service/userService.js

```diff
@@ -0,0 +1,75 @@
+import User from '../model/user'
+import * as userRepository from '../repository/userRepository'
+import { USER_ALREADY_EXISTS, USER_NOT_FOUND } from '../error/errorMessages'
+
+function createUser (userData) {
+  validateUniqueUserLogin(userData)
```

Comment:
Good temp solution but in real project these errors will be generated by DB and then by `mongoose, sequelize, typeorm` .etc if you set `login` as unique filed and you will have to make your controllers and services `async` to `await` db queries. So if you will do task with real db you can wrap controllers in function like this:

```
const catchAsync = (fn) => {
  return (req, res, next) => {
    fn(req, res, next).catch(next);
  };
};
```

Reply:
Thank you, I'll do it with real DB integration